packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/outbound.ts

```diff
@@ -46,6 +46,15 @@ async function defaultInvokeFunction(req: AWS.Lambda.InvocationRequest): Promise
   return lambda.invoke(req).promise();
 }
 
+async function defaultGetFunction(req: AWS.Lambda.GetFunctionRequest): Promise<AWS.Lambda.GetFunctionResponse> {
+  if (!lambda) {
+    lambda = new AWS.Lambda(awsSdkConfig);
+  }
+
+  return lambda.getFunction(req).promise();
+}
+
```

Comment:
So, according to the documentation, we can run into a situation where new lamba functions also haven't finished provisioning before we try to invoke them so I wonder if the right path forward here is to replace `defaultInvokeFunction with the following (roughly, this also contains notes):
```
async function defaultInvokeFunction(req: AWS.Lambda.InvocationRequest): Promise<AWS.Lambda.InvocationResponse> {
  if (!lambda) {
    lambda = new AWS.Lambda(awsSdkConfig);
  }

  let status = await defaultGetFunction(req);
  if (status.Configuration?.State !== 'Failed') {
    try {
      // If it is inactive, pending, or failed it will throw instead of returning
      return lambda.invoke(req).promise();
    } catch (e) {
      // We only care about inactive or pending for retries
      while (status.Configuration?.State === 'Inactive' || status.Configuration?.State === 'Pending') {
        // Add wait of some sort in here if we want
        // Since we don't know how long the reprovisioning of resources will take, we shouldn't add in a max time.
        status = await defaultGetFunction(req);
      }
    }
  }
  // Now that it's either active or failed, return the terminal response, error or not
  return lambda.invoke(req).promise();
}
```

This would mean that we don't want to export `defaultGetFunction` as `getFunction` since the scope of its use is only in this file. This edit would require no change (I think) to the logic in `invokeUserFunction()` in `framework.ts`. Well, except that there's still the issue with the ARN that I commented on. Maybe we just never take in qualified ARNs, though, so maybe that isn't an issue. On the other hand, you could export `getFunction` and use that to make sure we're getting the correct function name. It's part of `Configuration`.


What do you think?